### ISSUE TITLE
Any request to a different URL than baseUrl before cy.visit causes dev server to break

### ISSUE THREAD

**Issue by dflor003** [cypress-io/cypress#27186]
### Current behavior

If you make any requests (i.e. via `cy.request(...)`) to another URL that is not the `baseUrl` before calling `cy.visit(...)`, the dev server stops working and returns `ECONNREFUSED` for any subsequent requests.

Our specific use-case for this is that we have our backend as a separate app with its own URL/port. We passed an environment variable, `API_URL` to cypress and make requests using that as the base before we start any of our tests to seed test data.

### Desired behavior

Hitting the `baseUrl` should work regardless of any traffic to any other sites prior to calling `cy.visit('/')`

### Test code to reproduce

In theory, this should reproduce it:

```ts
describe('Some E2e Test', () => {
  it('should reproduce the issue', () => {
    cy.request('https://google.com');
    cy.visit('/'); // Should see ECONNREFUSED here
  });
});
```

Flipping the two lines should "fix" it, but it's not ideal in all cases.

### Cypress Version

12.14.0

### Node version

Local: 18.16.0, Bundled: 16.16.0

### Operating System

Windows 10.0.19045

**Comment by MikeMcC399** [comment]
@dflor003 

It sounds like you may be visiting more than one superdomain.

Would you like to check the [Web Security > Limitations](https://docs.cypress.io/guides/guides/web-security#Limitations) documentation to see if it applies to your case?

**Comment by dflor003** [comment]
Yeah, this seems like the case. I lost 2 days trying to figure out why this was breaking our tests. Could the error messaging around this from within Cypress be improved so others don't lose time to this? Perhaps cypress could throw an error message when you try to make requests to a different origin with a link to the documentation. Right now the behavior is that the original host just returns `ECONNREFUSED`.

**Comment by MikeMcC399** [comment]
@dflor003 

I'm sorry to hear that you spent much time on this! I'm not deeply enough into this aspect to judge whether it could be trapped with an error message. I can well understand that this is what you would like to see. We'll have to see what the Cypress.io team says. You are not the only person to be tripped up by this.

**Comment by nagash77** [comment]
I will move this ticket along to the backlog as a potential improvement.  Just to set realistic expectations, I do not think this will be a high priority and so any solution will likely require an outside contribution.
